### PR TITLE
sysvar: modify tidb_enable_historical_stats behavior

### LINKED ISSUE #30821: modify tidb_enable_historical_stats behavior
## Enhancement

In https://github.com/pingcap/tidb/pull/30646 we merged a new system variable: `tidb_enable_historical_stats`.

It uses the functions `getTiDBTableValue()` / `setTiDBTableValue()` to persist the saved values to the `mysql.tidb` table, but for new sysvars we don't need to do this. We should just rely on the built in saving/loading method.

The functions `getTiDBTableValue()` / `setTiDBTableValue()`  are unfortunately required for existing variables related to garbage collection. This is because we do not have clear rules about upgrade/downgrade and need to support earlier versions of TiDB which read/write from these tables and do not use system variables.

We can not clean this up these functions until product management to define support versions/upgrade rules (cc @easonn7 ), but it does not have a use for new sysvars.

The advantage of using the built-in method is that the values are cached on each tidb-server in memory. If we add too many rows to the `mysql.tidb` table it will affect performance, because when `SHOW VARIABLES` is executed each value must be read one row at a time.

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #30821 

Problem Summary:

We have a system variable cache based on etcd, so the getter and setter of `tidb_enable_historical_stats` are no longer depended on `mysql.tidb`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by CbcWestwolf** [comment]
PTAL @morgo @hawkingrei

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -1185,9 +1185,10 @@ var defaultSysVars = []*SysVar{
 		return config.GetGlobalConfig().Plugin.Dir, nil
 	}},
 	{Scope: ScopeGlobal, Name: TiDBEnableHistoricalStats, Value: Off, Type: TypeBool, GetGlobal: func(s *SessionVars) (string, error) {
-		return getTiDBTableValue(s, "tidb_enable_historical_stats", Off)
+		return s.GlobalVarsAccessor.GetGlobalSysVar(TiDBEnableHistoricalStats)
 	}, SetGlobal: func(s *SessionVars, val string) error {
-		return setTiDBTableValue(s, "tidb_enable_historical_stats", val, "Current historical statistics enable status")
+		SetSysVar(TiDBEnableHistoricalStats, val)
+		return nil
 	}},
```
Comment (morgo) [inline sessionctx/variable/sysvar.go:1192]: You can just remove the GetGlobal and SetGlobal funcs. This will be done automatically.

**Review by morgo — COMMENTED** [review]
Small change, otherwise LGTM

**Review by morgo — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**